yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1183,6 +1187,29 @@ object Client extends Logging {
     YarnSparkHadoopUtil.addPathToEnvironment(env, Environment.CLASSPATH.name, path)
 
   /**
+   * Returns the path to be sent to the NM for a local path string.
+   *
+   * This method uses two configuration values:
+   *
+   * - spark.yarn.config.localPath: a string that identifies a portion of the input path that may
+   *   only be valid in the local process.
+   * - spark.yarn.config.clusterPath: a string with which to replace the local path. This may
+   *   contain, for example, env variable references, which will be expanded by the NMs when
+   *   starting containers.
+   *
+   * If either config is not available, the input path is returned.
+   */
+  def getClusterPath(conf: SparkConf, path: String): String = {
```

Comment:
It's very tightly-coupled with all the code in this class that handles paths, and not really used anywhere else... perhaps when we clean up this code we can move all the path-handling code to a more appropriate location.